pallets/funding/src/tests/7_settlement.rs

```diff
@@ -1,445 +1,1382 @@
+			let bidder = first_bid.bidder;
+			inst.execute(|| {
+				assert_noop!(
+					crate::Pallet::<TestRuntime>::settle_successful_bid(
+						RuntimeOrigin::signed(bidder),
+						project_id,
+						bidder,
+						first_bid.id
+					),
+					Error::<TestRuntime>::IncorrectRound
+				);
+			});
+		}
+	}
+}
+
+#[cfg(test)]
+mod settle_successful_contribution_extrinsic {
+	use super::*;
+
+	#[cfg(test)]
+	mod success {
+		use super::*;
+
+		#[test]
+		fn contribution_is_correctly_settled() {
+			let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
+			let project_metadata = default_project_metadata(ISSUER_1);
+			let issuer = ISSUER_1;
+			let evaluations =
+				inst.generate_successful_evaluations(project_metadata.clone(), default_evaluators(), default_weights());
+			let bids = inst.generate_bids_from_total_ct_percent(
+				project_metadata.clone(),
+				50,
+				default_weights(),
+				default_bidders(),
+				default_multipliers(),
+			);
+			let mut community_contributions = inst.generate_contributions_from_total_ct_percent(
+				project_metadata.clone(),
+				40,
+				default_weights(),
+				default_community_contributors(),
+				default_community_contributor_multipliers(),
+			);
+
+			let contribution_mul_1 =
+				ContributionParams::<TestRuntime>::new(BUYER_6, 1000 * CT_UNIT, 1, AcceptedFundingAsset::USDT);
+			let contribution_mul_2 =
+				ContributionParams::<TestRuntime>::new(BUYER_7, 1000 * CT_UNIT, 2, AcceptedFundingAsset::USDT);
+
+			community_contributions.push(contribution_mul_1);
+
+			let project_id = inst.create_remainder_contributing_project(
+				project_metadata.clone(),
+				issuer,
+				evaluations,
+				bids,
+				community_contributions,
+			);
+			let wap = inst.get_project_details(project_id).weighted_average_price.unwrap();
+
+			let plmc_required = inst.calculate_contributed_plmc_spent(vec![contribution_mul_2.clone()], wap);
+			let plmc_ed = plmc_required.accounts().existential_deposits();
+			inst.mint_plmc_to(plmc_required.clone());
+			inst.mint_plmc_to(plmc_ed);
+
+			let usdt_required = inst.calculate_contributed_funding_asset_spent(vec![contribution_mul_2.clone()], wap);
+			inst.mint_foreign_asset_to(usdt_required.clone());
+
+			inst.execute(|| {
+				assert_ok!(PolimecFunding::remaining_contribute(
+					RuntimeOrigin::signed(BUYER_7),
+					get_mock_jwt_with_cid(
+						BUYER_7,
+						InvestorType::Professional,
+						generate_did_from_account(BUYER_7),
+						project_metadata.clone().policy_ipfs_cid.unwrap(),
+					),
+					project_id,
+					contribution_mul_2.amount,
+					contribution_mul_2.multiplier,
+					contribution_mul_2.asset
+				));
+			});
+
+			inst.finish_funding(project_id).unwrap();
+			assert_eq!(inst.get_project_details(project_id).status, ProjectStatus::FundingSuccessful);
+			let settlement_block = inst.get_update_block(project_id, &UpdateType::StartSettlement).unwrap();
+			inst.jump_to_block(settlement_block);
+
+			// First contribution assertions
```

Comment:
One has a vesting schedule, the other doesn't. I don't think it's worth abstracting into a loop for only 2 items to avoid duplicating 10 lines of code, if I'm going to have to add lines anyway to do vesting/no vesting assertions